crates/sui-core/src/batch_bls_verifier.rs

```diff
@@ -355,27 +369,27 @@ fn batch_verify_certificates_impl(
     obligation.verify_all()
 }
 
-// Cache up to 20000 verified certs. We will need to tune this number in the future - a decent
+// Cache up to 40000 verified digests. We will need to tune this number in the future - a decent
 // guess to start with is that it should be 10-20 times larger than peak transactions per second,
-// on the assumption that we should see most certs twice within about 10-20 seconds at most: Once via RPC, once via consensus.
-const VERIFIED_CERTIFICATE_CACHE_SIZE: usize = 20000;
+// on the assumption that we should see most certs and user signatures twice within about 10-20 seconds at most: Once via RPC, once via consensus.
+const VERIFIED_DIGEST_CACHE_SIZE: usize = 40000;
 
-pub struct VerifiedCertificateCache {
-    inner: RwLock<LruCache<CertificateDigest, ()>>,
+pub struct VerifiedDigestCache {
```

Comment:
Did we rename that for reusable LRU structs and we generalize it to digests-LRU?